pkg/ext-proc/backend/provider.go

```diff
@@ -80,9 +88,11 @@ func (p *Provider) Init(refreshPodsInterval, refreshMetricsInterval time.Duratio
 	go func() {
 		for {
 			time.Sleep(refreshMetricsInterval)
-			if err := p.refreshMetricsOnce(); err != nil {
+			ctx, cancel := context.WithTimeout(context.Background(), fetchMetricsTimeout)
```

Comment:
instead of doing this twice, here and at the beginning of the init function, move this to `refreshMetricsOnce` function.

Reply:
We need to create a new context with new timeout(deadline) every time we call refreshMetricsOnce.

Reply:
yes, and doing it inside `refreshMetricsOnce` will achieve that.

Reply:
right, was thinking asomething else :) fix is on the way